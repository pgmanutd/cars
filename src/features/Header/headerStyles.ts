import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';

export const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      borderBottom: `1px solid ${theme.palette.divider}`,
    },
    toolbar: {
      justifyContent: 'space-between',
    },
    toolbarLogo: {
      margin: theme.spacing(1, 1.5),
    },
    link: {
      margin: theme.spacing(1, 1.5),
    },
  }),
);
