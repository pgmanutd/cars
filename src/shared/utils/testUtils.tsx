import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory, MemoryHistory } from 'history';
import { render } from '@testing-library/react';

export const renderWithRouter = (
  ui: React.ReactElement,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  }: { route?: string; history?: MemoryHistory } = {},
) => ({
  ...render(<Router history={history}>{ui}</Router>),
  history,
});
