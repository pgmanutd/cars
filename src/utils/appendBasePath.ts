const appendBasePath = (basePath: string, path: string) =>
  `${basePath}?${path}`;

export default appendBasePath;
