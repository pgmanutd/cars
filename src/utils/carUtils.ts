import { Car } from 'features/CarInfo/types';

export const getCarName = ({ manufacturerName, modelName }: Car) =>
  `${manufacturerName} ${modelName}`;

export const getCarFeatures = (
  { stockNumber, mileage, fuelType, color }: Car,
  { stockNumberPrefix = '' }: { stockNumberPrefix?: string } = {
    stockNumberPrefix: '',
  },
) =>
  `${stockNumberPrefix}${stockNumber} - ${mileage?.number} ${mileage?.unit} - ${fuelType} - ${color}`;
