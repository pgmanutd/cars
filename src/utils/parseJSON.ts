const parseJSON = <T>(stringifiedJSON: string | null, fallbackValue: T) => {
  try {
    if (!stringifiedJSON) {
      return fallbackValue;
    }

    return JSON.parse(stringifiedJSON);
  } catch (e) {
    return fallbackValue;
  }
};

export default parseJSON;
