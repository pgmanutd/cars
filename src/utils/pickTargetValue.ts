import React from 'react';

const pickTargetValue = (event: React.ChangeEvent<{ value: unknown }>) =>
  event.target.value;

export default pickTargetValue;
