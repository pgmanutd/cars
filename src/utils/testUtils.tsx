import React from 'react';
import { Router, Route } from 'react-router-dom';
import { createMemoryHistory, MemoryHistory } from 'history';
import { HelmetProvider } from 'react-helmet-async';
import { render } from '@testing-library/react';

import { TranslateProvider } from 'features/Translate';

export const renderWithProviders = (
  ui: React.ReactElement,
  {
    routerConfig: {
      path = '/',
      route = '/',
      history = createMemoryHistory({ initialEntries: [route] }),
    },
  }: {
    routerConfig: { path?: string; route?: string; history?: MemoryHistory };
  } = { routerConfig: {} },
) => ({
  renderResult: render(
    <TranslateProvider>
      <HelmetProvider>
        <Router history={history}>
          <Route
            path={path}
            render={routeProps => React.cloneElement(ui, { ...routeProps })}
          />
        </Router>
      </HelmetProvider>
    </TranslateProvider>,
  ),
  history,
});

export const renderWithTranslate = (ui: React.ReactElement) => ({
  renderResult: render(<TranslateProvider>{ui}</TranslateProvider>),
});
