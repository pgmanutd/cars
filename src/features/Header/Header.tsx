import React from 'react';
import Link from '@material-ui/core/Link';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';

import routePaths from 'constants/routePaths';

import { useTranslate } from 'features/Translate';
import AppLogo from 'features/AppLogo';

import { useStyles } from './headerStyles';

const Header: React.FC = props => {
  const classes = useStyles();
  const { translate } = useTranslate();

  return (
    <AppBar
      data-testid="Header"
      {...props}
      color="default"
      elevation={0}
      className={classes.root}
    >
      <Toolbar className={classes.toolbar}>
        <Link
          color="textPrimary"
          href={routePaths.cars}
          className={classes.toolbarLogo}
        >
          <AppLogo />
        </Link>
        <nav>
          <Link color="textPrimary" href="#" className={classes.link}>
            {translate('features.Header.purchaseNavLinkText')}
          </Link>
          <Link
            color="textPrimary"
            href={routePaths.favorites}
            className={classes.link}
          >
            {translate('features.Header.myOrdersNavLinkText')}
          </Link>
          <Link color="textPrimary" href="#" className={classes.link}>
            {translate('features.Header.sellNavLinkText')}
          </Link>
        </nav>
      </Toolbar>
    </AppBar>
  );
};

export default Header;
