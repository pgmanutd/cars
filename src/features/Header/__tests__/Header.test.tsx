import React from 'react';

import { renderWithTranslate } from 'utils/testUtils';

import Header from '../Header';

describe('<Header />', () => {
  const setup = () => {
    const { renderResult } = renderWithTranslate(<Header />);

    return {
      renderResult,
    };
  };

  it('should render the component and matches it against stored snapshot', () => {
    const { renderResult } = setup();

    expect(renderResult.asFragment()).toMatchSnapshot();
  });
});
