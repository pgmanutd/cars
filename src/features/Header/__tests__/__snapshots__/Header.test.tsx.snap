// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the component and matches it against stored snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic makeStyles-root-1 MuiAppBar-colorDefault MuiPaper-elevation0"
    data-testid="Header"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-2 MuiToolbar-gutters"
    >
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-toolbarLogo-3 MuiTypography-colorTextPrimary"
        href="/"
      >
        <img
          alt="AppLogo"
          data-testid="AppLogo"
          src="logo.png"
        />
      </a>
      <nav>
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-4 MuiTypography-button MuiTypography-colorTextPrimary"
          href="#"
        >
          Purchase
        </a>
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-4 MuiTypography-button MuiTypography-colorTextPrimary"
          href="/favorites"
        >
          My Orders
        </a>
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-4 MuiTypography-button MuiTypography-colorTextPrimary"
          href="#"
        >
          Sell
        </a>
      </nav>
    </div>
  </header>
</DocumentFragment>
`;
